server/api.go

```diff
@@ -223,6 +223,33 @@ func (d dexAPI) ListPasswords(ctx context.Context, req *api.ListPasswordReq) (*a
 
 }
 
+func (d dexAPI) GetPassword(ctx context.Context, req *api.GetPasswordReq) (*api.GetPasswordResp, error) {
+	if req.Email == "" {
+		return nil, errors.New("no email supplied")
+	}
+
+	password, err := d.s.GetPassword(req.Email)
+	if err != nil {
+		if err == storage.ErrNotFound {
+			return &api.GetPasswordResp{
+				NotFound: true,
+			}, nil
+		}
+		d.logger.Errorf("api: there was an error retrieving the password: %v", err)
+		return nil, fmt.Errorf("get password: %v", err)
+	}
+
+	return &api.GetPasswordResp{
+		NotFound: false,
```

Comment:
[nit] this line is superfluous, but being explicit is OK 👍

Reply:
Yep was trying to be explicit.